src/Microsoft.TestPlatform.ObjectModel/Host/ITestRunTimeProvider.cs

```diff
@@ -63,6 +64,13 @@ public interface ITestRuntimeProvider
         void SetCustomLauncher(ITestHostLauncher customLauncher);
 
         /// <summary>
+        /// Gets the end point address and behaviour of TestRuntime
+        /// E.g. for phone device EndPoint:127.0.0.1:8080, ConnectionRole Host, TransportProtocol: Sockets
+        /// </summary>
+        /// <returns> Socket where the service is hosted by TestRuntime. Empty otherwise</returns>
+        ConnectionInfo GetTestHostConnectionInfo();
```

Comment:
Not sure if this API can ever return empty? Did we mean the ConnectionInfo.Endpoint will be empty?